doc/spec_TransferElement.dox

```diff
@@ -213,9 +232,49 @@ This documnent is currently still **INCOMPLETE**!
+  - 12.4 A TransferGroup is becomes read-only (cf. TransferGroup::isReadOnly()), if at least one of the member TransferElements is read-only.
+  - 12.5 Adding the same TransferElement twice is not allowed (logic_error).
+  - 12.6 Adding one TransferElement to two different TransferGroups is not allowed (logic_error).
+  - 12.7 Adding TransferElements with AccessMode::wait_for_new_data is not allowed (logic_error).
+  - \anchor transferElement_B_12_8 12.8 Adding a copy of the a TransferElement which is already in the group (returned by two different calls to Device::getZzzRegisterAccessor() resp. DeviceBackend::getRegisterAccessor() with the same arguments) will replace one of the identical TransferElement with a decorator which copies the data from the other.
+    - 12.8.1 This avoids an unnecessary transfer of identical data.
+    - \anchor transferElement_B_12_8_2 12.8.2 This is only possible, if the TransferElement is added by passing a TransferElementAbstractor to TransferGroup::addAccessor(). \ref transferElement_comment_B_12_8_2 "(*)"
+  - 12.9 Operations executed via TransferGroup::read() and TransferGroup::write():
+    - 12.9.1 first call preXxx() on all high-level elements (cf. \ref transferElement_B_12_1_2_1 "12.1.2.1"),
+    - 12.9.2 next call the corresponding xxxTransferYyy() on all low-level elements (cf. \ref transferElement_B_12_1_2_2 "12.1.2.2"), and
+    - 12.9.3 finally call postXxx() on all high-level elements.
+  - 12.10 The TransferGroup implements a similar exception handling in TransferGroup::read() and TransferGroup::write() as their TransferElement counterparts. This especially means:
+    - 12.10.1 Before preXxx() is called, the TransferGroup checks all pre-conditions (device opened and operation allowed) to make sure no ChimeraTK::logic_error can occur (other exceptions are anyway not allowed in preXxx).
+      - \anchor transferElement_B_12_10_1_1 12.10.1.1 If any ChimeraTK::runtime_error occurs while checking the pre-conditions, it is immediately passed on to the application.
+      - \anchor transferElement_B_12_10_1_2 12.10.1.2 Any pre-condition which is not met will immediately result in a thrown ChimeraTK::logic_error
+      - 12.10.1.3 The exceptions from \ref transferElement_B_12_10_1_1 "12.10.1.1" and \ref transferElement_B_12_10_1_2 "12.10.1.2" are not prioritised by type. The first detected error throws.
+    - 12.10.2 ChimeraTK::runtime_error exceptions thrown by the low-level elements are propagated to the corresponding high-level elements so they are seen by their postXxx() functions (c.f. \ref transferElement_B_6_3 "6.3")
+    - 12.10.3 postXxx() is called for all high level elements, even if some postXxx() throw exceptions (c.f. \ref transferElement_B_5 "5").
+      - 12.10.3.1 If exceptions have been caught from any postXxx(), an exception is thrown after all of them have been executed.
+      - 12.10.3.2 The thrown exception will be identical to the first thrown exception.
```

Comment:
Sounds confusion. What is "the first thrown exception"?
Proposal:
The first exception that was seen by the transfer group in 12.9.2 or 12.9.3 is re-thrown (*)

(*) "Was seen" means either as _activeException of an low level TE after calling the transfer step, or a bad_numeric_cast caught in postRead().

Reply:
I can do that, but I think you are merely confusing two implementations which have the same behaviour. You can either check which exception has appeared first in the transfer phase in _activeException of the low-level elements, or you can build some smart try-catch block around the postXxx() calls and rethrow the first thrown exception. Your suggestion implies the first implementation, while just stating "first thrown exception" is neutral to this.

If I am wrong about this: what else do you think this can be confused with?

Reply:
(I actually find your suggestion more confusing... It's overcomplicating a simple idea and implies a specific implementation.)

Reply:
Well, the comment might be confusing. And actually, to implement it correctly you have to remember the first _activeExeption in the transfer phase (only those have the correct order) AND you need the try catch block for the bad_numeric_cast.

But that's not what I don't like. The wording is not clear.
```
The thrown exception will be identical to the first thrown exception.
```
I don't like the first *thrown* exception, especially after "the thrown exception". It's the first caught exception when I am inside the group code. And the "will be identical to" also does not help to make it clearer. In fact, it is the same exception, thrown with std::rethrow_exception.

How about this:
```
The first exception that has been caught by the transfer group in 12.9.2 or 12.9.3 is re-thrown.
```
without the comment. We could in fact call readTransfer() directly and catch in the group, and not call handleTransferException and look at _activeException. But that's implementation detail, it does not belong into the spec.

Reply:
Ok I will change it to:

> The first exception that has been caught in 12.9.2 or 12.9.3 is re-thrown.

Then it's neutral w.r.t. the implementation.

Reply:
12.9.2 is for runtime_error and 12.9.3 is only for numeric_cast error, right?